src/appBootstrapper.jsx

```diff
@@ -21,6 +23,27 @@ const stores = {
     // ...other stores
 };
 
+const end = superagent.Request.prototype.end;
+
+let redirecting = false;
+
+superagent.Request.prototype.end = function (callback) {
+    return end.call(this, (error, response) => {
+        if (redirecting) {
+            return;
+        }
+        if (response.statusCode === 401) {
+            var storageKey = `redirect${Math.floor(Math.random() * 1000000000000)}`
```

Comment:
We want to make sure that after redirecting to login page and logging in, we restore the hashfragment that was created when the redirect was initiated, and not some other hashfragment that could possible have gotten stuck in local storage (e.g. failed login attempt).  Then unique key just ensures that will never happen.